curp/src/rpc/mod.rs

```diff
@@ -169,85 +186,100 @@ impl ProposeResponse {
             }
-            Some(ExeResult::Error(ref buf)) => Ok(failure(ProposeError::decode(buf)?)),
+            Some(ExeResult::Error(ref err)) => {
+                let propose_error = err
+                    .clone()
+                    .propose_error
+                    .ok_or(PbSerializeError::EmptyField)?
+                    .try_into()?;
+                Ok(failure(propose_error))
+            }
             None => Ok(success(None)),
         }
     }
 }
 
 impl WaitSyncedRequest {
     /// Create a `WaitSynced` request
-    pub(crate) fn new(id: &ProposeId) -> bincode::Result<Self> {
-        Ok(Self {
-            id: bincode::serialize(id)?,
-        })
+    pub(crate) fn new(id: &ProposeId) -> Self {
+        Self {
+            propose_id: id.clone().into_inner(),
+        }
     }
 
     /// Get the propose id
-    pub(crate) fn id(&self) -> bincode::Result<ProposeId> {
-        bincode::deserialize(&self.id)
+    pub(crate) fn propose_id(&self) -> ProposeId {
+        ProposeId::new(self.propose_id.clone())
     }
 }
 
 impl WaitSyncedResponse {
     /// Create a success response
-    pub(crate) fn new_success<C: Command>(asr: &C::ASR, er: &C::ER) -> bincode::Result<Self> {
-        Ok(Self {
+    pub(crate) fn new_success<C: Command>(asr: &C::ASR, er: &C::ER) -> Self {
+        Self {
             sync_result: Some(SyncResultRaw::Success(Success {
-                after_sync_result: bincode::serialize(&asr)?,
-                exe_result: bincode::serialize(&er)?,
+                after_sync_result: asr.encode(),
+                exe_result: er.encode(),
             })),
-        })
+        }
     }
 
     /// Create an error response
-    pub(crate) fn new_error<C: Command>(err: &CommandSyncError<C>) -> bincode::Result<Self> {
-        Ok(Self {
-            sync_result: Some(SyncResultRaw::Error(bincode::serialize(&err)?)),
-        })
+    pub(crate) fn new_error<C: Command>(err: CommandSyncError<C>) -> Self {
+        Self {
+            sync_result: Some(SyncResultRaw::Error(PbCommandSyncErrorOuter {
+                command_sync_error: Some(err.into()),
+            })),
+        }
     }
 
     /// Create a new response from execution result and `after_sync` result
     pub(crate) fn new_from_result<C: Command>(
         er: Option<Result<C::ER, C::Error>>,
         asr: Option<Result<C::ASR, C::Error>>,
-    ) -> bincode::Result<Self> {
+    ) -> Self {
         match (er, asr) {
             (None, Some(_)) => {
                 unreachable!("should not call after sync if execution fails")
             }
             (None, None) => WaitSyncedResponse::new_error::<C>(
-                &SyncError::Other("can't get er result".to_owned()).into(),
+                SyncError::Other("can't get er result".to_owned()).into(),
             ), // this is highly unlikely to happen,
             (Some(Err(_)), Some(_)) => {
                 unreachable!("should not call after_sync when exe failed")
             }
```

Comment:
Please unify the content of these two `unreachable`, They have the same meaning. but use different sentences

Reply:
done